src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -358,6 +359,60 @@ def _prepare_for_mds_upgrade(
 
         return continue_upgrade
 
+    def _post_upgrade_redeploys(self, target_digests: List[str]) -> bool:
```

Comment:
I think we can vastly simplify this patch and drop this method entirely, and instead just add the constraint that ``deployed_by`` match to the code in ``_do_upgrade`` that generates the ``to_upgrade`` list.  AFAICS the flow to redeploy (for upgrade) and redeploy (because mgr version is different) is identical, except that the latter doesn't *need* to (but can!) pass the target image to ``_daemon_action``.  Then we don't need any new health check code, or pretty much anything except for that added check when building the list...

*Maybe*, for clarity's sake, we make the log messages not say ``Upgrading`` for those daemons?  Otherwise, it would be much simpler?

Reply:
I think you're right. I'm pretty sure the separate function was here from earlier attempts at this (before using unit.meta file) where it wasn't feasible to do inside the normal upgrade loop and I never thought to redo it. I'll try to fit it into the existent loop instead. The one question is where in the upgrade should the monitoring stack daemons be done since they'd have to be added to the existing upgrade order to be part of the current upgrade loop. If we're going to have a later change that prevents reconfigs while the daemon doesn't have the right "deployed_by" like discussed in standup maybe we should just tack them on to the end?